gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -523,26 +524,36 @@ static void checkValidFloatingPoint(double value) {
    * @since 2.2
    */
   public <T> TypeAdapter<T> getDelegateAdapter(TypeAdapterFactory skipPast, TypeToken<T> type) {
-    // Hack. If the skipPast factory isn't registered, assume the factory is being requested via
-    // our @JsonAdapter annotation.
-    if (!factories.contains(skipPast)) {
-      skipPast = jsonAdapterFactory;
-    }
-
-    boolean skipPastFound = false;
-    for (TypeAdapterFactory factory : factories) {
-      if (!skipPastFound) {
-        if (factory == skipPast) {
+    if (skipPast != null) {
```

Comment:
This is necessary because `jsonAdapterFactory.isFieldAdapterFactory` would throw a NullPointerException otherwise.

This slightly changes the behavior in that previously `null` was treated as jsonAdapterFactory, which was however not documented and also likely not intended.